packages/wallet/frontend/src/pages/transactions.tsx

```diff
@@ -269,32 +331,88 @@ const TransactionsPage: NextPageWithLayout<TransactionsPageProps> = ({
       {!error && !loading ? (
-        <div className="flex w-full items-center justify-between">
-          <Button
-            className="disabled:pointer-events-none disabled:from-gray-400 disabled:to-gray-500"
-            aria-label="go to previous page"
-            disabled={Number(pagination.page) - 1 < 0}
-            onClick={() => {
-              const previousPage = Number(pagination.page) - 1
-              if (isNaN(previousPage) || previousPage < 0) return
-              redirect({ page: previousPage.toString() })
-            }}
-          >
-            Previous
-          </Button>
-          <Button
-            className="disabled:pointer-events-none disabled:from-gray-400 disabled:to-gray-500"
-            aria-label="go to next page"
-            disabled={Number(pagination.page) + 1 > totalPages - 1}
-            onClick={() => {
-              const nextPage = Number(pagination.page) + 1
-              if (isNaN(nextPage) || nextPage > totalPages - 1) return
-              redirect({ page: nextPage.toString() })
-            }}
-          >
-            Next
-          </Button>
-        </div>
+        <>
+          <div className="flex w-full items-center justify-between">
+            <Button
+              className="hidden disabled:pointer-events-none disabled:from-gray-400 disabled:to-gray-500 md:flex"
+              aria-label="go to previous page"
+              disabled={Number(pagination.page) - 1 < 0}
+              onClick={() => {
+                const previousPage = Number(pagination.page) - 1
+                if (isNaN(previousPage) || previousPage < 0) return
+                redirect({ page: previousPage.toString() })
+              }}
+            >
+              Previous
+            </Button>
+            {totalPages !== 1 && (
+              <div className="flex w-full justify-center">
+                <IconButton
+                  className="mx-3 md:hidden"
+                  aria-label="go back"
+                  onClick={() => {
+                    const previousPage = Number(pagination.page) - 1
+                    if (isNaN(previousPage) || previousPage < 0) return
+                    redirect({ page: previousPage.toString() })
+                  }}
+                >
+                  <Play className="h-4 w-4 rotate-180" />
+                </IconButton>
+                {pages.map((page) => {
+                  if (
+                    Math.abs(page - 1 - Number(pagination.page)) <= 1 ||
+                    page === 1 ||
+                    page === totalPages
+                  ) {
+                    return (
+                      <li key={page} className="list-none p-1">
+                        <Button
+                          size="xs"
+                          intent="outline"
+                          className={cx(
+                            page - 1 === Number(pagination.page) &&
+                              '!bg-orange text-white'
+                          )}
```

Comment:
Let's make the buttons green as well. I feel like the white-orange combination is a bit challenging for the eyes.